src/api/java/com/minecolonies/api/entity/mobs/AbstractEntityMinecoloniesMob.java

```diff
@@ -329,7 +366,14 @@ public IColony getColony()
      */
     public void registerWithColony()
     {
-        getColony();
+        if (colony == null || eventID == 0)
```

Comment:
Good idea

Reply:
ye all raiders now have to know their colony and eventID and are removed otherwise. The events gonna take care to spawn a new one if needed for the event